File: Python/OOP/MathDojo.py
class MathDojo(object):
    def __init__(self):
        self.result = 0

    def add(self, *args):
        self.result += self._add(args)
        return self

    def subtract(self, *args):
        self.result -= self._add(args)
        return self

    def _add(self, array):
        sum = 0
        for value in array:
            if isinstance(value, tuple) or isinstance(value, list):
                sum += self._add(value)
            else:
                sum += value
        return sum

print MathDojo().add(2).add(2, 5, (2,(2,50))).subtract(3, 2).result

File: Ruby/first/ruby_exercises/math_dojo.rb
class MathDojo
  attr_reader :result

  def initialize
    @result = 0;
  end
  def add *params
    i = 0
    while i < params.length do
      if params[i].class == [].class
        params += params[i]
      else
        @result += params[i]
      end
      i += 1
    end
    return self
  end

  def subtract *params
    i = 0
    while i < params.length do
      if params[i].class == [].class
        params += params[i]
      else
        @result -= params[i]
      end
      i += 1
    end
    return self
  end
end

math = MathDojo.new
puts math.add([4, [5, 9]], 2).subtract([4, [5, 9]], 2).result
